src/Fixer/ListNotation/ListSyntaxFixer.php

```diff
@@ -0,0 +1,163 @@
+                    array('syntax' => 'long')
+                ),
+                new VersionSpecificCodeSample(
+                    "<?php\nlist(\$sample) = \$array;",
+                    new VersionSpecification(70100),
+                    array('syntax' => 'short')
+                ),
+            )
+        );
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getPriority()
+    {
+        // should be run before the BinaryOperatorSpacesFixer and TernaryOperatorSpacesFixer.
+        return 1;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function isCandidate(Tokens $tokens)
+    {
+        return PHP_VERSION_ID >= 70100 && $tokens->isTokenKindFound($this->candidateTokenKind);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function applyFix(\SplFileInfo $file, Tokens $tokens)
+    {
+        $callback = $this->fixCallback;
+        for ($index = $tokens->count() - 1; 0 <= $index; --$index) {
+            if ($tokens[$index]->isGivenKind($this->candidateTokenKind)) {
+                $this->$callback($tokens, $index);
+            }
+        }
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function createConfigurationDefinition()
+    {
+        $syntax = new FixerOptionBuilder('syntax', 'Whether to use the `long` or `short` `list` syntax.');
+        $syntax = $syntax
+            ->setAllowedValues(array('long', 'short'))
+            ->setDefault('long')
+            ->getOption()
+        ;
+
+        return new FixerConfigurationResolver(array($syntax));
+    }
+
+    /**
+     * @param Tokens $tokens
+     * @param int    $index
+     */
+    private function fixToLongSyntax(Tokens $tokens, $index)
+    {
+        static $typesOfInterest = array(
+            array(CT::T_DESTRUCTURING_SQUARE_BRACE_CLOSE),
+            array(CT::T_ARRAY_SQUARE_BRACE_OPEN),
+        );
+
+        $closeIndex = $tokens->getNextTokenOfKind($index, $typesOfInterest);
+        if (!$tokens[$closeIndex]->isGivenKind(CT::T_DESTRUCTURING_SQUARE_BRACE_CLOSE)) {
+            return;
+        }
+
+        $tokens->overrideAt($index, '(');
+        $tokens->overrideAt($closeIndex, ')');
+
+        $tokens->insertAt($index, new Token(array(T_LIST, 'list')));
+    }
+
+    /**
+     * @param Tokens $tokens
+     * @param int    $index
+     */
+    private function fixToShortSyntax(Tokens $tokens, $index)
+    {
+        $openIndex = $tokens->getNextTokenOfKind($index, array('('));
+        $closeIndex = $tokens->findBlockEnd(Tokens::BLOCK_TYPE_PARENTHESIS_BRACE, $openIndex);
+
+        $tokens->overrideAt($openIndex, array(CT::T_DESTRUCTURING_SQUARE_BRACE_OPEN, '['));
+        $tokens->overrideAt($closeIndex, array(CT::T_DESTRUCTURING_SQUARE_BRACE_CLOSE, ']'));
+
+        $tokens->clearTokenAndMergeSurroundingWhitespace($index);
+    }
+
+    private function resolveFixCallback()
+    {
+        $this->fixCallback = sprintf('fixTo%sSyntax', ucfirst($this->configuration['syntax']));
```

Comment:
i know this approach was already used in other fixer, but i really don't like it. static code analysis with things like this is horrible